packages/integrations/cloudflare/src/index.ts

```diff
@@ -291,24 +298,47 @@ export default function createIntegration(args?: Options): AstroIntegration {
 					const entryPaths = entryPointsURL.map((entry) => fileURLToPath(entry));
 					const outputUrl = new URL('$astro', _buildConfig.server);
 					const outputDir = fileURLToPath(outputUrl);
-
-					await esbuild.build({
-						target: 'es2020',
-						platform: 'browser',
-						conditions: ['workerd', 'worker', 'browser'],
-						entryPoints: entryPaths,
-						outdir: outputDir,
-						allowOverwrite: true,
-						format: 'esm',
-						bundle: true,
-						minify: _config.vite?.build?.minify !== false,
-						banner: {
-							js: SHIM,
-						},
-						logOverride: {
-							'ignored-bare-import': 'silent',
-						},
-					});
+					//
+					// Sadly, this needs to build esbuild for each depth of routes/entrypoints independently so that relative
```

Comment:
@alexanderniebuhr could you explain why, please? I'm curious :)

Reply:
I'll add my 2 cents. It would be nice if this was zero configuration. Its not really "changing" behavior like other configs, rather its adding behavior that the user would expect to work given what works in cloudflare.

That being said, this multiple esbundle depending on depth of the entrypoint is a bit of a hack, and if it stays that way, adding a configuration to opt-in to wasm loading for at least the function per route case makes sense to me.

@alexanderniebuhr were your reservations around this hack or something else?

About the hack, In my development, I wasn't able to specify a different import pattern for the different entrypoints using its `onResolve` callback. But it seems like there must be some other better way to do it. I was just hoping that someone might have better insight. Looking deeper at the esbuild plugin documentation, I might be able to do something with the [`onEnd` callback](https://esbuild.github.io/plugins/#on-end) to modify the imports within each entry

Reply:
@ematipico @adrianlyjak let me explain!

I'm genuinely excited about this feature and would love to see it enabled by default in the long run. 🚀

However, I've got a few thoughts:

- I'm a bit hesitant about the `entryPathsGroupedByDepth` solution. I believe there might be a more efficient approach out there. I'm already thinking about it but need some more time, to fully understand the requirements.
- However I don't think it should block this PR - we should iterate on it in a future PR!
- Given that `wasm` isn't a typical use-case for majority of users, my suggestion would be to make this feature opt-in for now. This way, we ensure we don't inadvertently degrade the experience for users without `wasm` - which we test for, but it still feels a bit hacky.
- I also haven't dived deeper into a thorough code review yet. So I don't really know if the current solution is that hacky or fine. I was just going by my initial gut feeling and the comment provided by @adrianlyjak.
- Once we've tested this with real-world use cases & optimized the adapter to reduce code branches _(I'm working on that)_, I would consider making the default `true`, but still leave the flag intact _(for user choice)_.

This mirrors how I usually approach runtime support: opt-in first & switching to default once I'm sure it works for 95% of users.

Reply:
Yeah, I'm looking a little deeper at the hack, and I think I may have just overlooked some things that should be able to remove the `entryPathsGroupedByDepth`. I'll let you know once that's updated.

I've also implemented a flag. That makes sense to slow roll this in case there is potential any potential fallout from the new functionality.

Reply:
> However I don't think it should block this PR - we should iterate on it in a future PR!

Well, the [onEnd](https://esbuild.github.io/plugins/#on-end) callback looks promising, but still doesn't appear trivial to implement, so perhaps yes, given the opt-in, deferring that optimization seems like a better decision.

I'm currently running into another head scratcher. This doesn't work well in hybrid mode where a client side page imports wasm. I've added a failing test.

Currently, due to the vite build, it's just rewriting to a `*.wasm` import for cloudflare workers. However, when generating the static pages, that doesn't work, giving the following opaque error.

```
[vite:wasm-module-loader] Could not load /Users/adrianlyjak/dev/futz/astro/packages/integrations/cloudflare/test/fixtures/wasm/src/util/add.wasm?module (imported by test/fixtures/wasm/src/pages/hybrid.ts): WASM module's cannot be loaded unless you add `wasmModuleImports: true` to your astro config.
```

I tried tweaking the vite/rollup plugin. The SSG build isn't in a node-like environment, but I can embed the wasm as a base64 string. The unfortunate thing here though is it seems like the SSG mode and SSR mode use the same vite build, so the vite build is somewhat at a stand still. Cloudflare only supports importing .wasm modules, and the ssg builder does not. And unfortunately this only shows up at build time, so the dev server is misleading that things will work.

The only further idea to support both would be to use base64 in the vite build always, and then somehow re-interpret an wasm import within the worker esbuild.

Reply:
@adrianlyjak I'm keen to assist and offer a fresh perspective. While I haven't fully grasped all the issues yet, I'll delve deeper into your implementation to better understand the challenges you've described. I believe there might be some hooks we can leverage to streamline the process. To reiterate my earlier point, I feel we should merge this PR as it stands and focus on refining it in subsequent iterations.

What are your thoughts, @ematipico?

Reply:
> The only further idea to support both would be to use base64 in the vite build always, and then somehow re-interpret an wasm import within the worker esbuild.

I ended up doing this. It seems to work well and not be too complicated. Main complaint is that there's loosely coupled and hard to follow assumptions between the 2 build stages

Reply:
Given the long discussion and the technical details that surfaced, I think we should keep the feature opt-in, so we can discuss them offline (discord, github issue, etc.) I want @adrianlyjak 's contribution to be merged ASAP, but we shouldn't rush into "enabled by default".

Reply:
Currently the feature is opt-in. What's the process for merging? Or do you want further code review? I see this is starting to accumulate conflicts, and I'm hoping to not have to keep up with those. I'll rebase this for the time being (excited about the node compatibility that's conflicting here 😄 )

Reply:
For me it looks good!
In general the process is other reviews, including docs. Once they are approve, merging will be handled by one of the maintainers :)